packages/core/package.json

```diff
@@ -44,7 +44,7 @@
       "require": "./dist/index.js"
     },
     "./rsc": {
-      "types": "./rsc/dist/index.d.ts",
+      "types": "./rsc/dist/index.d.mts",
```

Comment:
does this work with typescript / commonjs setups?

Reply:
Since this subpath only has ESM builds, so no CJS related issues. We have a shared .mjs file in dist and the import must be `import ... from './mod.mjs'` because as per the spec, the extension can't be omitted. Then, due to https://www.typescriptlang.org/docs/handbook/modules/reference.html#the-moduleresolution-compiler-option, when TS sees that `./mod.mjs` it will look for `.d.mts` instead of `.d.ts`.